moped-editor/src/views/projects/projectView/ProjectTeam/ProjectTeamTable.js

```diff
@@ -0,0 +1,506 @@
+import TeamAutocompleteComponent from './TeamAutocompleteComponent';
+import DataGridTextField from 'src/components/DataGridPro/DataGridTextField';
+
+const useStyles = makeStyles((theme) => ({
+  infoIcon: {
+    fontSize: "1rem",
+    verticalAlign: "sub",
+    color: theme.palette.text.primary,
+    "&:hover": {
+      color: theme.palette.primary.main,
+    },
+  },
+  roleHeader: {
+    fontWeight: 500,
+  },
+}));
+
+const useTeamNameLookup = (data) => 
+  useMemo(() => {
+    if (!data) {
+      return {};
+    }
+    return data.moped_users.reduce((obj, item) => {
+      obj[item.user_id] = `${item.first_name} ${item.last_name}`;
+      return obj;
+    }, {});
+  }, [data]);
+
+const useRoleNameLookup = (data) => 
+  useMemo(() => {
+    if (!data) {
+      return {};
+    }
+    return data.moped_project_roles.reduce((obj, item) => {
+      obj[item.project_role_id] = item.project_role_name;
+      return obj;
+    }, {});
+  }, [data]);
+
+const useColumns = ({
+  data,
+  rowModesModel,
+  handleEditClick,
+  handleSaveClick,
+  handleCancelClick,
+  handleDeleteOpen,
+  classes,
+  teamNameLookup,
+  roleNameLookup
+}) => 
+  useMemo(() => {
+    return [
+  { 
+    headerName: 'Name', 
+    field: 'moped_user', 
+    width: 250,
+    editable: true,
+    valueGetter: (user) => {
+      return user ? `${user.first_name} ${user.last_name}` : '';
+    },
+    renderEditCell: (props) => {
+      return(
+      <TeamAutocompleteComponent
+        {...props}
+        name={"user"}
+        value={props.row.moped_user}
+        nameLookup={teamNameLookup}
+        />
+      );
+    }
+  },
+  { 
+    headerName: 'Workgroup', 
+    field: 'moped_workgroup',
+    width: 200,
+    valueGetter: (workgroup) => workgroup?.workgroup_name,
+  },
+  { 
+    headerName: 'Role',
+    field: 'moped_proj_personnel_roles', 
+    width: 200,
+    editable: true,
+    renderHeader: () => (
+      <div className={classes.roleHeader}>
+        Role{" "}
+        <Link
+          href="https://atd-dts.gitbook.io/moped/user-guides/project-team"
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          <Icon className={classes.infoIcon}>info_outline</Icon>
+        </Link>
+      </div>
+    ),
+    renderCell: (params) => {
+      // Filter out deleted roles and map to Typography components
+      const roleElements = params.row.moped_proj_personnel_roles
+        .filter(role => !role.is_deleted)
+        .map(role => (
+          <Typography key={role.moped_project_role?.project_role_id} sx={{ fontSize: "0.875rem" }}>
+            {role.moped_project_role?.project_role_name}
+          </Typography>
+        ));
+      
+      return <Box>{roleElements}</Box>;
+    },
+    renderEditCell: (props) => {
+      return (
+        <ProjectTeamRoleMultiselect
+          {...props}
+          value={props.row.moped_proj_personnel_roles || []}
+          roles={data.moped_project_roles}
+        />
+      );
+    },
```

Comment:
i noticed that i cannot save a row without this input being populated, but there is no validation feedback when i try to save when this value is null. might be a nice enhancement.